Yoast/javascript
# Add ability to pass refs to buttons and buttonstyledlinks

## Summary

Special projects needed to call a button via its ref, which is currently not possible. Therefore, we've added a `buttonRef` prop, to which a `ref` can be forwarded by e.g. `React.forwardRef()`;


This PR can be summarized in the following changelog entry:

* Button and ButtonStyledLink now accept an optional buttonRef prop, which will be set as a ref on the html button and anchor element, respectively.


## Test instructions

This PR can be tested by following these steps:

* Build components app, go to the `Reactified Components` tab, and click the example buttons. The `This Button has a Ref!` and `This ButtonStyledLink has a Ref!` buttons should focus upon clicking.


## UI changes

## Quality assurance


Fixes #
